src/coreNatives.cpp

```diff
@@ -17,23 +17,26 @@
 
 #include "spmod.hpp"
 
-// native void printToServer(const char[] text, any ...)
-static cell_t printToServer(SourcePawn::IPluginContext *ctx,
+// native void PrintToServer(const char[] text, any ...)
+static cell_t PrintToServer(SourcePawn::IPluginContext *ctx,
                                  const cell_t *params)
 {
     char *formatString;
     char bufferOutput[1024];
 
     ctx->LocalToString(params[1], &formatString);
-    gSPGlobal->formatString(bufferOutput, sizeof(bufferOutput), formatString, ctx, params, 2);
-
+    size_t res = gSPGlobal->formatString(bufferOutput, sizeof(bufferOutput), formatString, ctx, params, 2);
+  
+    bufferOutput[res++] = '\n';
+    bufferOutput[res] = '\0';
```

Comment:
not sure its necessary, but its exists in both sourcemod & amxx

Reply:
Just for sanity.

Buffer overflow can occur here. You should subtract ```2``` from buffer size like in the [SourceMod](https://github.com/alliedmodders/sourcemod/blob/master/core/logic/smn_console.cpp#L96) or make a condition like in the [AMX Mod X](https://github.com/alliedmodders/amxmodx/blob/master/amxmodx/amxmodx.cpp#L102).

Reply:
Ah, sure, thats why we need string terminator